src/Hermes.package/HEFloat.class/instance/doReadFrom..st

```diff
@@ -1,8 +1,11 @@
 reading
 doReadFrom: aReader
-	value := BoxedFloat64 new.
-	value at:1 put: (aReader readUInt32).
-	value at:2 put: (aReader readUInt32).
+	| originalValue |
+	originalValue := BoxedFloat64 new.
+	originalValue at:1 put: (aReader readUInt32).
+	originalValue at:2 put: (aReader readUInt32).
 	
 	"Force the conversion to SmallFloat64 or BoxedFloat64"
-	value := value + 0
\ No newline at end of file
+	value := originalValue + 0.
+	
+	(value literalEqual: originalValue) ifFalse:[ value := value negated]
```

Comment:
writing value := originalValue - 0.0. would work for negative zero and avoid this sign discussion dance.

Reply:
Do you know if this behaviour is consistent in all the platforms?

Reply:
Yes, it's covered by ANSI/IEEE Std 754–1985, IEEE Standard for Binary Floating-Point Arithmetic and its successors, in §6.3 The sign bit:

>When the sum of two operands with opposite signs (or the difference of two operands with like signs) is exactly zero, the sign of that sum (or difference) shall be + in all rounding modes except round toward –INFINITY, in which mode that sign shall be –. However, x+x = x–(–x) retains the same sign as x even when x is zero

Hence (-0.0) - (0.0) which has to be interpreted same as (-0.0) + (-0.0) is necessarily -0.0 (same sign as x operand...)

A platform that would not behave consistently would mean a problem of compiler and/or compiler options and should be fixed ASAP.